worlds/metroidprime/data/RoomData.py

```diff
@@ -0,0 +1,464 @@
+                    paired_door = target_room_data.get_matching_door(door_data, world)
+                    shield_applied = False
+                    if (
+                        paired_door
+                        and paired_door.blast_shield
+                        and paired_door.blast_shield != BlastShieldType.No_Blast_Shield
+                    ):
+                        door_data.blast_shield = paired_door.blast_shield
+                        shield_applied = True
+                    elif (
+                        paired_door
+                        and door_data.blast_shield
+                        and paired_door.blast_shield != BlastShieldType.No_Blast_Shield
+                    ):
+                        paired_door.blast_shield = door_data.blast_shield
+                        shield_applied = True
+
+                    if shield_applied:
+                        door_data.lock = DoorLockType.Blue
+
+                def sub_region_access_rule_func(
+                    state: CollectionState,
+                    world: "MetroidPrimeWorld",
+                    origin_door_data: DoorData,
+                    target_door_data: DoorData,
+                ):
+                    meets_origin_door_requirements = (
+                        origin_door_data.sub_region_access_override(world, state)
+                        and self._can_open_door(world, state, origin_door_data)
+                        if origin_door_data.sub_region_access_override is not None
+                        else self._can_access_door(world, state, origin_door_data)
+                    )  # Use override if any, otherwise use default access rule
+                    return meets_origin_door_requirements and self._can_open_door(
+                        world, state, target_door_data
+                    )
+
+                def get_connection_name(
+                    door_data: DoorData,
+                    target_room_name: str = name,
+                    target_destination: RoomName = destination,
+                ) -> str:
+                    if door_data.blast_shield:
+                        pass
+                    blast_shield_text = (
+                        ""
+                        if door_data.blast_shield is None
+                        or door_data.blast_shield == BlastShieldType.No_Blast_Shield
+                        else f" {door_data.blast_shield.value}"
+                    )
+                    lock = door_data.lock or door_data.defaultLock
+                    return (
+                        lock.value
+                        + blast_shield_text
+                        + f" Door from {target_room_name} to {target_destination.value}"
+                    )
+
+                apply_blast_shield_to_both_sides_of_door(door_data)
+
+                target_region = world.get_region(
+                    door_data.get_destination_region_name()
+                )
+                entrance = region.connect(

```

Comment:
An additional benefit to pre-determining door rules is that `BlastShieldType.Disabled` doors are impassable, and can skip connecting regions in the first place.